cfme/infrastructure/config_management.py

```diff
@@ -338,21 +579,21 @@ def __init__(self, name, profile, appliance=None):
 
     def tag(self, tag):
         """Tags the system by given tag"""
-        navigate_to(self, 'EditTags')
-        fill(mixins.tag_form, {'category': 'Cost Center *', 'tag': 'Cost Center 001'})
-        # ---
-        mixins.add_tag(tag, navigate=False)
+        view = navigate_to(self, 'EditTags')
+        view.form.fill({'category': 'Cost Center *', 'tag': 'Cost Center 001'})
+        view.form.save.click()
 
     def untag(self, tag):
         """Removes the selected tag off the system"""
-        navigate_to(self, 'EditTags')
-        mixins.remove_tag(tag)
+        view = navigate_to(self, 'EditTags')
+        row = view.form.table.row(category=tag.category, name=tag.name)
+        row.actions.click()
 
     @property
     def tags(self):
         """Returns a list of this system's active tags"""
-        navigate_to(self, 'EditTags')
-        return mixins.get_tags()
+        view = navigate_to(self, 'EditTags')
+        return [(r.category, r.assigned_value) for r in view.entities.table]
```

Comment:
**REQUIRED**
These `tag` method should be removed and `ConfigSystem` should inherit `WidgetasticTaggable` class.